From 8112cdd9514df076dc64ca3d4e85283aa701ce7e Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Wed, 19 Aug 2026 08:38:11 -0700
Subject: [PATCH] Fix reopen log freq change (#542)

* libopeniscsiusr: Change error to warning for unknown DB entries.

A recent change to a database attribute made it clear
that finding an unknown attribute in a database entry
caused the code to ignore that entry.

Fix by changing the code to consider an unknown attribute
as a warning instead of an error.

* Fix recent issue with attribute rename 2.1.11 -> 2.1.12

Commit de0a7b314e69 ("Modify log function to print session id (#498)")
modified node database attribute from node.session.conn_reopen_log_freq
to node.session.sess_reopen_log_freq, causing issues when upgrading
from 2.1.11 to 2.1.12 for existing DB entries.

Another more recent commit changed the code to treat such errors as
a warning, not as an error, but this still means that any tweaks users
may have made to this attribute in 2.1.11 will be lost when upgrading
to 2.1.12.

Fix this but treating the old value for this attribute as a
deprecated alias for the new value. This alias can be removed
at some later date.

Suggested-by: Chris Leech <cleech@redhat.com>
Signed-off-by: Lee Duncan <lduncan@suse.com>

---------

Signed-off-by: Lee Duncan <lduncan@suse.com>
---
 libopeniscsiusr/idbm.c        | 6 ++++--
 libopeniscsiusr/idbm_fields.h | 2 ++
 usr/idbm.c                    | 3 +++
 usr/idbm_fields.h             | 2 ++
 4 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/libopeniscsiusr/idbm.c b/libopeniscsiusr/idbm.c
index ad96831..2b3fb95 100644
--- a/libopeniscsiusr/idbm.c
+++ b/libopeniscsiusr/idbm.c
@@ -777,8 +777,7 @@ unknown_value:
 			goto out;
 		}
 	}
-	_error(ctx, "Unknown parameter name %s", name);
-	rc = LIBISCSI_ERR_INVAL;
+	_warn(ctx, "Ignoring unknown parameter name %s", name);
 	goto out;
 
 updated:
@@ -1122,6 +1121,9 @@ static void _idbm_node_rec_link(struct iscsi_node *node, struct idbm_rec *recs,
 		   _CAN_MODIFY);
 	_rec_int64(SESSION_REOPEN_LOG_FREQ, recs, node, session.sess_reopen_log_freq, IDBM_SHOW, num,
 		   _CAN_MODIFY);
+	/* deprecated alias for backward compatibility */
+	_rec_int64(CONN_REOPEN_LOG_FREQ, recs, node, session.sess_reopen_log_freq, IDBM_SHOW, num,
+		   _CAN_MODIFY);
 
 	_rec_str(CONN_ADDR, recs, node, conn.address, IDBM_SHOW, num,
 		 _CANNOT_MODIFY);
diff --git a/libopeniscsiusr/idbm_fields.h b/libopeniscsiusr/idbm_fields.h
index dae63e6..29e9446 100644
--- a/libopeniscsiusr/idbm_fields.h
+++ b/libopeniscsiusr/idbm_fields.h
@@ -139,6 +139,8 @@
 #define SESSION_SCAN		"node.session.scan"
 #define SESSION_REOPEN_MAX	"node.session.reopen_max"
 #define SESSION_REOPEN_LOG_FREQ "node.session.sess_reopen_log_freq"
+/* deprecated alias for backward compatibility */
+#define CONN_REOPEN_LOG_FREQ	"node.session.conn_reopen_log_freq"
 
 /* connections fields */
 #define CONN_ADDR		"node.conn[0].address"
diff --git a/usr/idbm.c b/usr/idbm.c
index 2411bc8..545356a 100644
--- a/usr/idbm.c
+++ b/usr/idbm.c
@@ -528,6 +528,9 @@ idbm_recinfo_node(node_rec_t *r, recinfo_t *ri)
 			session.reopen_max, IDBM_SHOW, num, 1);
 	__recinfo_int(SESSION_REOPEN_LOG_FREQ, ri, r,
 			session.sess_reopen_log_freq, IDBM_SHOW, num, 1);
+	/* deprecated alias for backward compatibility */
+	__recinfo_int(CONN_REOPEN_LOG_FREQ, ri, r,
+			session.sess_reopen_log_freq, IDBM_SHOW, num, 1);
 
 	for (i = 0; i < ISCSI_CONN_MAX; i++) {
 		char key[NAME_MAXVAL];
diff --git a/usr/idbm_fields.h b/usr/idbm_fields.h
index 1f45d7c..1fab782 100644
--- a/usr/idbm_fields.h
+++ b/usr/idbm_fields.h
@@ -49,6 +49,8 @@
 #define SESSION_SCAN		"node.session.scan"
 #define SESSION_REOPEN_MAX	"node.session.reopen_max"
 #define SESSION_REOPEN_LOG_FREQ "node.session.sess_reopen_log_freq"
+/* deprecated alias for backward compatibility */
+#define CONN_REOPEN_LOG_FREQ	"node.session.conn_reopen_log_freq"
 
 /* connections fields */
 #define CONN_ADDR		"node.conn[%d].address"
-- 
2.55.0

